moor/lib/src/dsl/columns.dart

```diff
@@ -165,6 +168,15 @@ class ColumnBuilder<Builder, ResultColumn extends Column<ResultDartType>,
   /// `String`, which would usually be used for [Table.text] columns.
   Builder map<T>(TypeConverter<T, ResultDartType> converter) => _isGenerated();
 
+  ///
+  Builder foreignKey(
+    Type references, {
+    KeyAction onUpdate = KeyAction.noAction,
+    KeyAction onDelete = KeyAction.noAction,
+    String column = 'id',
```

Comment:
I think this shouldn't have a default value. We could also make it a `Symbol`. The we could let users write something like

```dart
.foreignKey(Categories, #id, onUpdate: ...);
```

Reply:
Symbol is a good advice. What parameters should'n be default? I think the referenced (primary key) column is "id" in the cases of 99%. That's why I made it default. Also, - for example - I never provide "onUpdate" in most cases, because i know id fields won't be changed. SQL implementation also not requires ON UPDATE/DELETE clauses in references definition (that means no action). But this is not made just for me :) If you say this fields should be required, I accept.

Reply:
But yes, it can be a nice declaration with two positional argument and two named. Looks like the resulting sql statement. :)

Reply:
`onUpdate` and `onDelete` have defaults in sqlite3 (`NO ACTION`), so it's reasonable to copy them too.

My problem with `id` being the default is that it might be confusing for users declaring tables with a different primary key. They might assume that, since they didn't have to declare the column in most other cases, moor was somehow smart enough to just pick the right column all of the time :) So far we're not, so it's best to just make it explicit.